src/SignalR/common/Http.Connections.Common/src/NegotiateProtocol.cs

```diff
@@ -5,230 +5,244 @@
+                            }
+                            else
+                            {
+                                Skip(ref reader);
+                            }
+                            break;
+                        case JsonTokenType.EndObject:
+                            completed = true;
+                            break;
+                        default:
+                            throw new InvalidDataException($"Unexpected token '{reader.TokenType}' when reading negotiation response JSON.");
                     }
+                }
 
-                    if (url == null && error == null)
+                if (url == null && error == null)
+                {
+                    // if url isn't specified or there isn't an error, connectionId and available transports are required
+                    if (connectionId == null)
                     {
-                        // if url isn't specified or there isn't an error, connectionId and available transports are required
-                        if (connectionId == null)
-                        {
-                            throw new InvalidDataException($"Missing required property '{ConnectionIdPropertyName}'.");
-                        }
-
-                        if (availableTransports == null)
-                        {
-                            throw new InvalidDataException($"Missing required property '{AvailableTransportsPropertyName}'.");
-                        }
+                        throw new InvalidDataException($"Missing required property '{ConnectionIdPropertyName}'.");
                     }
 
-                    return new NegotiationResponse
+                    if (availableTransports == null)
                     {
-                        ConnectionId = connectionId,
-                        Url = url,
-                        AccessToken = accessToken,
-                        AvailableTransports = availableTransports,
-                        Error = error,
-                    };
+                        throw new InvalidDataException($"Missing required property '{AvailableTransportsPropertyName}'.");
+                    }
                 }
+
+                return new NegotiationResponse
+                {
+                    ConnectionId = connectionId,
+                    Url = url,
+                    AccessToken = accessToken,
+                    AvailableTransports = availableTransports,
+                    Error = error,
+                };
             }
             catch (Exception ex)
             {
                 throw new InvalidDataException("Invalid negotiation response received.", ex);
             }
         }
 
-        private static AvailableTransport ParseAvailableTransport(JsonTextReader reader)
+        public static NegotiationResponse ParseResponse(Stream content)
+        {
+            var buffer = new byte[content.Length];
```

Comment:
Is it guaranteed for the stream to be seekable here? If so, add `Debug.Assert(content.CanSeek)`? Also, do we need other arg validation like a null check?

Reply:
I was thinking of marking this function as obsolete.

Reply:
Even being marked as obsolete, this implementation seems worse than just throwing a NotImplementedException. Even if the Stream is seekable and has a Length, I don't think it's guaranteed the entire content is buffered and ready to go. Stream.Read() could block or return only partial content.

Do we include calls to this method in any of our samples, demos or docs?

Reply:
I'm not sure why this class is public, I'm guessing it was a mistake.

Reply:
It's not that big of a deal, hopefully it won't matter in the long run. Mark it obselete and we can carry on with our lives.